guidance/models/_model.py

```diff
@@ -17,11 +17,19 @@
 import numpy as np
 
 try:
+    from IPython import get_ipython
     from IPython.display import clear_output, display, HTML
-
-    ipython_is_imported = True
 except ImportError:
     ipython_is_imported = False
+    notebook_mode = False
+else:
+    ipython_is_imported = True
+    _ipython = get_ipython()
+    notebook_mode = (
+        _ipython is not None
+        and "IPKernelApp" in _ipython.config
```

Comment:
Based on the way that `tqdm.auto` determines notebook context -- would be good to test this on multiple machines/platforms...

Reply:
We had to tackle this detection back in InterpretML too, here's how we did it there: https://github.com/interpretml/interpret/blob/develop/python/interpret-core/interpret/provider/_environment.py. We could re-use some of this logic here perhaps

cc @nopdive

Reply:
Sorry, hadn't been watching my notifications -- code seems reasonable. Yes, this should be tested (either manually/automatically is fine) on multiple platforms before merge: terminal (Window/Linux/Mac), vscode, jupyter notebook/lab, azure/google/amazon/databricks notebooks. We should be relatively okay if these target environments work.